docker/tarfile/src.go

```diff
@@ -335,7 +335,7 @@ func (s *Source) prepareLayerData(tarManifest *ManifestItem, parsedConfig *manif
 		if err != nil {
 			return nil, err
 		}
-		if li, ok := unknownLayerSizes[h.Name]; ok {
+		if li, ok := unknownLayerSizes[path.Clean(h.Name)]; ok {
```

Comment:
Can  you do this once and save the variable to be used below.  I am not sure how much path.Clean does, but we probably don't want to call it multiple times, if we don't have to.

Reply:
done